tests/components/bang_olufsen/test_media_player.py

```diff
@@ -1127,3 +1204,133 @@ async def test_async_browse_media(
     assert response["success"]
 
     assert (child in response["result"]["children"]) is present
+
+
+@pytest.mark.parametrize(
+    ("group_members", "expand_count", "join_count"),
+    [
+        # Valid member
+        ([TEST_MEDIA_PLAYER_ENTITY_ID_2], 1, 0),
+        # Touch to join
+        ([], 0, 1),
+    ],
+)
+async def test_async_join_players(
+    hass: HomeAssistant,
+    mock_mozart_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    mock_config_entry_2: MockConfigEntry,
+    group_members: list[str],
+    expand_count: int,
+    join_count: int,
+) -> None:
+    """Test async_join_players."""
+
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+
+    source_change_callback = (
+        mock_mozart_client.get_source_change_notifications.call_args[0][0]
+    )
+
+    # Add another entity
+    mock_config_entry_2.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry_2.entry_id)
+
+    # Set the source to a beolink expandable source
+    source_change_callback(BangOlufsenSource.TIDAL)
+
+    await hass.services.async_call(
+        "media_player",
+        "join",
+        {
+            ATTR_ENTITY_ID: TEST_MEDIA_PLAYER_ENTITY_ID,
+            ATTR_GROUP_MEMBERS: group_members,
+        },
+        blocking=True,
+    )
+
+    assert mock_mozart_client.post_beolink_expand.call_count == expand_count
+    assert mock_mozart_client.join_latest_beolink_experience.call_count == join_count
+
+
+@pytest.mark.parametrize(
+    ("source", "group_members", "expected_result", "error_type"),
+    [
+        # Invalid source
+        (
+            BangOlufsenSource.LINE_IN,
+            [TEST_MEDIA_PLAYER_ENTITY_ID_2],
+            pytest.raises(HomeAssistantError),
```

Comment:
Shouldn't this be `pytest.raises(ServiceValidationError)`? `ServiceValidationError` is a subclass of `HomeAssistantError`, I think that's why test passes.

Reply:
Yes. Fixed now